### ISSUE TITLE
n_jobs(num_threads) affect model accuracy

### ISSUE THREAD

**Issue by therickli** [Microsoft/LightGBM#2090]
## Environment info

Operating System: ubuntu16.04

CPU/GPU model: i7 8700

C++/Python/R version: Python3.7.2

LightGBM version or commit hash: 2.2.3

## Steps to reproduce

```
param = {
    "max_bin": 80,
    "num_leaves": 30,
    "min_child_samples": 35,
    "subsample": 0.8,
    "subsample_freq": 2,
    "colsample_bytree": 0.8,
    "reg_lambda": 0.3,
    "min_split_gain": 0.1,
    "learning_rate": 0.05,
    "max_depth": 9,
    "n_estimators": 200,
    "n_jobs": 2,
}
gbm = lgb.LGBMClassifier(**param)
gbm.fit(X_train, y_train)
y_pred = gbm.predict(X)
report = metrics.classification_report(y, y_pred)
print(report)
```
When n_jobs=1  the results are:

>               precision    recall  f1-score   support
> 
>            0       1.00      0.99      0.99    639987
>            1       0.29      0.49      0.37      5034

> AUC of PR curve = 0.340

When n_jobs=2  the results are:

>               precision    recall  f1-score   support
> 
>            0       1.00      1.00      1.00    639987
>            1       0.47      0.46      0.46      5034
> 

> AUC of PR curve = 0.378

When n_jobs=4  the results are:

>               precision    recall  f1-score   support
> 
>            0       1.00      0.99      0.99    639987
>            1       0.29      0.51      0.37      5034

> AUC of PR curve = 0.345

Sorry I can't provide the training and test data. 
I want to know why n_jobs would affect model accracy.

**Comment by guolinke** [comment]
refer to question 6 in https://github.com/Microsoft/LightGBM/blob/master/docs/FAQ.rst
